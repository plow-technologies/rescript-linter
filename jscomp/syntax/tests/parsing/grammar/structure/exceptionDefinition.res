exception ExitEarly
exception ExitEarly(int)
exception ExitEarly({x: int})
exception ExitEarly({"jsExit": int})
exception ExitEarly({@attr "jsExit": int})
exception ExitEarly({@attr "jsExit": int,})
exception ExitEarly({@attr "jsExit": int, @attr "code": int,})
exception ExitEarly({"jsExit": int},)
exception ExitEarly({"jsExit": int}, {"code": int})
exception ExitEarly({"jsExit": int}, int, {"code": int},)
exception ExitEarly(
  {@attr "jsExit": int, @attr "code": int,},
  {@attr "jsExit": int, @attr "code": int,},
)
exception ExitJsStyle({..})
exception ExitJsStyle({.. "code": int})
exception ExitJsStyle({.. "code": int,})
exception ExitJsStyle({.. @attr "code": int})
exception ExitJsStyle({.. @attr "code": int,})
exception ExitJsStyle({.. "code": int, "time": int,})
exception ExitJsStyle({.. @attr "code": int, @attr "time": int,})

@onConstructor
exception ExitEarly
@onConstructor
exception ExitEarly(int)

exception Exit = Terminate 
exception Exit = Lib.Terminate 
exception Exit = Ns.Lib.Terminate 

@onConstructor
exception Exit = Terminate 
@onConstructor
exception Exit = Lib.Terminate 
