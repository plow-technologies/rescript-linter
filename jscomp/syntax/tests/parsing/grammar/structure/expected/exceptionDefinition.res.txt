exception ExitEarly 
exception ExitEarly of int 
exception ExitEarly of {
  x: int } 
exception ExitEarly of < jsExit: int   >  
exception ExitEarly of < jsExit: int [@attr ]  >  
exception ExitEarly of < jsExit: int [@attr ]  >  
exception ExitEarly of < jsExit: int [@attr ] ;code: int [@attr ]  >  
exception ExitEarly of < jsExit: int   >  
exception ExitEarly of < jsExit: int   >  * < code: int   >  
exception ExitEarly of < jsExit: int   >  * int * < code: int   >  
exception ExitEarly of < jsExit: int [@attr ] ;code: int [@attr ]  >  *
  < jsExit: int [@attr ] ;code: int [@attr ]  >  
exception ExitJsStyle of < .. >  
exception ExitJsStyle of < code: int   ;.. >  
exception ExitJsStyle of < code: int   ;.. >  
exception ExitJsStyle of < code: int [@attr ]  ;.. >  
exception ExitJsStyle of < code: int [@attr ]  ;.. >  
exception ExitJsStyle of < code: int  ;time: int   ;.. >  
exception ExitJsStyle of < code: int [@attr ] ;time: int [@attr ]  ;.. >  
exception ExitEarly [@onConstructor ]
exception ExitEarly of int [@onConstructor ]
exception Exit = Terminate
exception Exit = Lib.Terminate
exception Exit = Ns.Lib.Terminate
exception Exit[@onConstructor ] = Terminate
exception Exit[@onConstructor ] = Lib.Terminate